packages/expect/src/types.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+import {Config} from '@jest/types';
+
+export type SyncExpectationResult = {
+  pass: boolean;
+  message: () => string;
+};
+
+export type AsyncExpectationResult = Promise<SyncExpectationResult>;
+
+export type ExpectationResult = SyncExpectationResult | AsyncExpectationResult;
+
+export type RawMatcherFn = (
+  expected: any,
+  actual: any,
+  options: any,
+) => ExpectationResult;
+
+export type ThrowingMatcherFn = (actual: any) => void;
+export type PromiseMatcherFn = (actual: any) => Promise<void>;
+
+type Tester = (a: any, b: any) => boolean | undefined;
```

Comment:
compare `Tester` and `MatcherState` to types exported from `Matchers.ts` in `jest-types` package

Reply:
I've actually pulled these types from the jest-types package. As in I've literally copy pasted them. I assumed that was something to do since the TODO says migrate the types to expect